ocaml/tests/test_tracing.ml

```diff
@@ -0,0 +1,58 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+open Tracing
+
+module D = Debug.Make (struct let name = "test_tracing" end)
+
+open D
+
+let _ =
+  set_default ~tags:[] ~endpoints:["bugtool"] ~processors:[] ~filters:[]
+    ~service_name:"xapi"
+
+let start_test_span () =
+  set ~name_label:"default" ~status:true () ;
+  let tracer = get_tracer ~name:"test_tracer" in
+  let span = Tracer.start ~tracer ~name:"test_task" ~parent:None () in
+  set ~name_label:"default" ~status:false () ;
+  span
+
+let test_hashtbl_leaks () =
+  let leak =
+    let span = start_test_span () in
+    match span with
+    | Ok x ->
+        if Tracer.assert_span_hashtbl_empty () then
+          true
+        else
+          let _ = Tracer.finish x in
+          Unix.sleep 30 ;
```

Comment:
Having large sleeps in unit test code is usually not a good idea, why is this exactly 30 seconds, can it be made smaller?
If there is a timeout somewhere else then the unit test should set that timeout to a small value for testing purposes (to something small like 0.2s, and wait a bit more like 0.3 or 0.4s)

Reply:
It's to wait for the export thread in tracing.ml to clear the span hashtable -we could add another function that force-exports without doing the wait and calling that instead?

Reply:
resolved in ebd6aead72445ef2c9af5090ca9432e065ca5fd6